src/main/java/org/openelisglobal/notebook/dao/NoteBookDAOImpl.java

```diff
@@ -19,6 +19,59 @@ public NoteBookDAOImpl() {
         super(NoteBook.class);
     }
 
+    @Override
+    @SuppressWarnings("unchecked")
+    public List<NoteBook> filterNoteBooks(List<NoteBookStatus> statuses, List<String> types, List<String> tags,
+            Date fromDate, Date toDate) {
+
+        StringBuilder hql = new StringBuilder("select distinct nb from NoteBook nb ");
+        hql.append("left join nb.tags t where 1=1 ");
```

Comment:
To fetch the notebooks aka notebook templates ,you must add a filter. `where nb.isTemplate = true ` otherwise it will return to you everything including both notebook templates and their entries which am sure you dont want